Compiler/src/typeinfer.jl

```diff
@@ -400,7 +400,7 @@ function inline_cost_model(interp::AbstractInterpreter, result::InferenceResult,
     end
 end
 
-function transform_result_for_local_cache(interp::AbstractInterpreter, result::InferenceResult)
+function transform_result_for_local_cache(interp::AbstractInterpreter, result::InferenceResult, edges::SimpleVector)
```

Comment:
Why were `edges::SimpleVector` added as an additional argument? Is it for consistency with `transform_result_for_cache`? I expected edges not to be relevant since there we're caching locally only, not globally.

Reply:
It would be forbidden and impossible for any code to use the global cache if any part of the intermediate results fail to perfectly track and record their edges

Reply:
I see, should we therefore adjust our implementation of `transform_result_for_local_cache`, or do we already track edges in another way?

(what confuses me is that the `edges::SimpleVector` argument was added while the implementation was not updated)

Reply:
The default implementation does not use the result, so it doesn't need the edges, but it seemed unreasonable to assume no overload would ever want to use result